Robust.Client/Audio/Sources/BaseAudioSource.cs

```diff
@@ -188,7 +188,7 @@ public float Gain
             if (!IsEfxSupported)
             {
                 AL.GetSource(SourceHandle, ALSourcef.Gain, out var priorGain);
-                priorOcclusion = priorGain / _gain;
+                priorOcclusion = _gain == 0 ? 1f : priorGain / _gain;
```

Comment:
```suggestion
                priorOcclusion = _gain == 0 ? 0f : priorGain / _gain;
```

I think this is more appropriate because priorGain and _gain would be close to 0.

Reply:
That would leave us in the same position wouldn't it? As the gain we set would remain 0.

Reply:
This is going to go from full volume to potentially 0 volume which seems worse than going the other way, no?

Reply:
Yeah. But even with 0 I still get the problem:
```
time: $1701817152194 hashcode 42355014 handle 2408  Old gain: 1 for file /Audio/Announcements/announce.ogg new gain 1 value recived: 0.63095737
time: $1701817152195 after 0.63095737
time: $1701817152195 hashcode 42355014 handle 2408  Old gain: 0.63095737 for file /Audio/Announcements/announce.ogg new gain 1 value recived: 0
time: $1701817152195 after 0
[DEBG] system.audio: Starting audio /Audio/Announcements/announce.ogg on entity 1418 with Volume -2 and gain 0 and pos <0, 0>
time: $1701817152198 hashcode 42355014 handle 2408  Old gain: 0 for file /Audio/Announcements/announce.ogg new gain 0 value recived: 0.63095737
time: $1701817152198 after 0
time: $1701817152208 hashcode 42355014 handle 2408  Old gain: 0 for file /Audio/Announcements/announce.ogg new gain 0 value recived: 0.63095737
time: $1701817152208 after 0
```
Here are some of the logs for announce for example. (Ignore the mess I was getting lost)
Initially gain is set according to the volume IIRC which is -2. 
after that we set it to 0.
From that point on since we take the prior gain which is 0 and multiply by what ever new value we got. We still get 0 (Or NaN)
That's at least how it is on my machine. I'm really confused on how it works on other machines. Or I'm missing the point entirely somehow

Reply:
multiplying 0 by 0 is still going to be 0, this piece of code would never hit NaN as it's not dividing by 0 unless your system is deeply broken.

Reply:
That's my point. So if we set prior occlusion to zero gain would remain zero forever since it always multiples by it when setting it

Reply:
When I wrote NaN there that is what happens currently. Without any changes. Since after setting the gain to zero once dividing by prior gain goes to NaN and it gets stuck that way.

Reply:
Asked my friend to check on her system (linux) And it seems that (ALC_EXT_EFX) is just not supported on mac. I guess if we want to leave it pretty close like 0.2 or 0.5? I'll do some local tests and see how that sounds

Reply:
I think 0.5 Sounds okay